src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/14.diff.sql

```diff
@@ -0,0 +1,424 @@
+    @endResourceSurrogateId bigint,
+    @batchSize int
+AS
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    DELETE Top(@batchSize) FROM dbo.ResourceWriteClaim
+    WHERE ResourceSurrogateId >= @startResourceSurrogateId AND ResourceSurrogateId < @endResourceSurrogateId
+
+    COMMIT TRANSACTION
+
+    return @@rowcount
+GO
+
+
+/*************************************************************
+    Stored procedures for batch delete ResourceParams
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     BatchDeleteResourceParams
+--
+-- DESCRIPTION
+--     Batch delete ResourceParams
+--
+-- PARAMETERS
+--     @tableName
+--         * Resource params table name
+--     @resourceTypeId
+--         * Resource type id
+--     @startResourceSurrogateId
+--         * The start ResourceSurrogateId
+--     @endResourceSurrogateId
+--         * The end ResourceSurrogateId
+--     @batchSize
+--         * Max batch size for delete operation
+CREATE OR ALTER PROCEDURE dbo.BatchDeleteResourceParams
+    @tableName nvarchar(128),
+    @resourceTypeId smallint,
+    @startResourceSurrogateId bigint,
+    @endResourceSurrogateId bigint,
+    @batchSize int
+AS
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    DECLARE @Sql NVARCHAR(MAX);
+    DECLARE @ParmDefinition NVARCHAR(512);
+
+    SET @sql = N'DELETE TOP(@BatchSizeParam) FROM ' + OBJECT_NAME(OBJECT_ID(@tableName)) + N' WHERE ResourceTypeId = @ResourceTypeIdParam AND ResourceSurrogateId >= @StartResourceSurrogateIdParam AND ResourceSurrogateId < @EndResourceSurrogateIdParam'
+    SET @parmDefinition = N'@BatchSizeParam int, @ResourceTypeIdParam smallint, @StartResourceSurrogateIdParam bigint, @EndResourceSurrogateIdParam bigint'; 
+
+    EXECUTE sp_executesql @sql, @parmDefinition,
+                          @BatchSizeParam = @batchSize,
+                          @ResourceTypeIdParam = @resourceTypeId,
+                          @StartResourceSurrogateIdParam = @startResourceSurrogateId,
+                          @EndResourceSurrogateIdParam = @endResourceSurrogateId
+
+    COMMIT TRANSACTION
+
+    return @@rowcount
+GO
+
+/*************************************************************
+    Stored procedures for disable indexes
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     DisableIndexes
+--
+-- DESCRIPTION
+--     Stored procedures for disable indexes
+--
+-- PARAMETERS
+--     @indexes 
+--         * indexes table
+CREATE OR ALTER PROCEDURE [dbo].[DisableIndexes]
+    @indexes dbo.IndexTableType_1 READONLY
+AS
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    DECLARE commands CURSOR FAST_FORWARD
+    FOR SELECT N'ALTER INDEX ' + QUOTENAME(indexes.IndexName) + ' ON ' + OBJECT_NAME(OBJECT_ID(indexes.TableName)) + ' Disable;'
```

Comment:
Small nit: OBJECT_NAME() doesn't return the schema, so if we were to have multiple schemas with tables with the same name, this might be ambiguous.  Not a blocker now but something to consider for future code maintainability.

Reply:
Thanks for the information, we change to check tablename before use. Then we can directly use @table without object_name. Thanks!